VSR/Sync: Remove jump_sync_target() kludge for R=2

sentientwaffle opened the pull request:
Thanks to https://github.com/tigerbeetle/tigerbeetle/pull/1346, there is no risk in a R=2 cluster of the leading replica overwriting a prepare required by the lagging replica. This workaround is not needed anymore!

(I removed the `"Cluster: sync: checkpoint diverges, sync (primary diverges)"` test in a separate commit to emphasize that the test still passes, it just isn't really needed anymore since it doesn't exercise anything interesting anymore.)

sentientwaffle committed:
VSR/Sync: Remove jump_sync_target() kludge for R=2

Thanks to https://github.com/tigerbeetle/tigerbeetle/pull/1346, there is no risk in a R=2 cluster of the leading replica overwriting a prepare required by the lagging replica. This workaround is not needed anymore!

(Removing the `"Cluster: sync: checkpoint diverges, sync (primary diverges)"` test in a separate commit to emphasize that the test still passes, it just isn't really needed anymore since it doesn't exercise anything interesting anymore.)

matklad commented on src/vsr/replica_test.zig:
```diff
@@ -964,49 +964,6 @@ test "Cluster: sync: sync, bump target, sync" {
     try TestReplicas.expect_sync_done(t.replica(.R_));
 }
 
-test "Cluster: sync: R=2" {
```
Hm, can we keep the test in? Even if there's no longer any special code needed, this is still an interesting case to test

sentientwaffle replied:
Sure! I made a couple tweaks to it:
- The test doesn't actually trigger state sync anymore. (R=2 clusters _can't_ trigger state sync anymore, not counting standbys.)
- Prepare more ops into the new checkpoint, to emphasize the pipeline.

sentientwaffle committed:
VSR/Sync: Modernize old test

This test no longer uses state sync.
And it requires no "special" code paths to work; it works automatically thanks to `vsr_checkpoint_interval`.
